src/vendor/github.com/xelabs/go-mysqlstack/sqlparser/ast.go

```diff
@@ -931,25 +995,25 @@ type LengthScaleOption struct {
 	Scale  *SQLVal
 }
 
-// ColumnKeyOption indicates whether or not the given column is defined as an
+// ColumnPrimaryKeyOption and ColumnUniqueKeyOption indicates whether or not the given column is defined as an
 // index element and contains the type of the option
-type ColumnKeyOption int
+type ColumnPrimaryKeyOption int
+type ColumnUniqueKeyOption int
 
 const (
-	// ColKeyNone enum.
-	ColKeyNone ColumnKeyOption = iota
+	// ColKeyPrimaryNone enum.
+	ColKeyPrimaryNone ColumnPrimaryKeyOption = iota
 
 	// ColKeyPrimary enum.
 	ColKeyPrimary
+)
 
-	// ColKeyUnique enum.
-	ColKeyUnique
+const (
+	// ColUniqueKeyNone enum.
+	ColKeyUniqueNone ColumnUniqueKeyOption = iota
 
 	// ColKeyUniqueKey enum.
 	ColKeyUniqueKey
-
-	// ColKey enum.
```

Comment:
it can't be deleted?

Reply:
In this place, key is used as primary key, and I had do the optimize for it for `[[PRIMARY] KEY]` next. But we don`t change the table structure and semantics.
```
column_definition:
    data_type [NOT NULL | NULL] [DEFAULT default_value]
      [AUTO_INCREMENT] [UNIQUE [KEY]] [[PRIMARY] KEY]
      [COMMENT 'string']
      [COLLATE collation_name]
      [COLUMN_FORMAT {FIXED|DYNAMIC|DEFAULT}]
      [STORAGE {DISK|MEMORY}]
      [reference_definition]
```